File: src/main/java/kr/co/lunatalk/global/config/security/CustomAuthenticationEntryPoint.java
package kr.co.lunatalk.global.config.security;

import com.fasterxml.jackson.databind.ObjectMapper;
import com.fasterxml.jackson.databind.SerializationFeature;
import com.fasterxml.jackson.datatype.jsr310.JavaTimeModule;
import jakarta.servlet.ServletException;
import jakarta.servlet.http.HttpServletRequest;
import jakarta.servlet.http.HttpServletResponse;
import kr.co.lunatalk.global.common.response.ErrorResponse;
import kr.co.lunatalk.global.common.response.GlobalResponse;
import kr.co.lunatalk.global.exception.ErrorCode;
import lombok.RequiredArgsConstructor;
import org.springframework.http.HttpStatus;
import org.springframework.security.core.AuthenticationException;
import org.springframework.security.web.AuthenticationEntryPoint;
import org.springframework.stereotype.Component;

import java.io.IOException;

@Component
@RequiredArgsConstructor
public class CustomAuthenticationEntryPoint implements AuthenticationEntryPoint {
	private  final ObjectMapper mapper;

	@Override
	public void commence(HttpServletRequest request, HttpServletResponse response, AuthenticationException authException) throws IOException, ServletException {
		response.setContentType("application/json;charset=UTF-8");
		response.setStatus(HttpStatus.UNAUTHORIZED.value());
//		ObjectMapper mapper = new ObjectMapper().registerModule(new JavaTimeModule()).disable(SerializationFeature.WRITE_DATES_AS_TIMESTAMPS);
		GlobalResponse errorResponse = GlobalResponse.fail(ErrorCode.AUTH_UNAUTHORIZED.getHttpStatus().value(), new ErrorResponse(ErrorCode.AUTH_UNAUTHORIZED.name(), ErrorCode.AUTH_UNAUTHORIZED.getMessage()));
		response.getWriter().write(mapper.writeValueAsString(errorResponse));
	}
}

File: src/main/kotlin/kr/co/lunatalk/global/config/security/CustomAuthenticationEntryPoint.kt
package kr.co.lunatalk.global.config.security

import com.fasterxml.jackson.databind.ObjectMapper
import jakarta.servlet.http.HttpServletRequest
import jakarta.servlet.http.HttpServletResponse
import kr.co.lunatalk.global.common.response.ErrorResponse
import kr.co.lunatalk.global.common.response.GlobalResponse
import kr.co.lunatalk.global.exception.ErrorCode
import kr.co.lunatalk.global.util.CookieUtil
import org.springframework.context.annotation.Lazy
import org.springframework.http.HttpStatus
import org.springframework.security.core.AuthenticationException
import org.springframework.security.web.AuthenticationEntryPoint
import org.springframework.stereotype.Component

@Component
class CustomAuthenticationEntryPoint(
    @Lazy private val mapper: ObjectMapper,
    @Lazy private val cookieUtil: CookieUtil
) : AuthenticationEntryPoint {

    override fun commence(
        request: HttpServletRequest,
        response: HttpServletResponse,
        authException: AuthenticationException
    ) {
        cookieUtil.deleteAccessTokenCookie(response)

        response.contentType = "application/json;charset=UTF-8"
        response.status = HttpStatus.UNAUTHORIZED.value()
        val errorResponse = GlobalResponse.fail(
            ErrorCode.AUTH_TOKEN_EXPIRED.httpStatus.value(),
            ErrorResponse(ErrorCode.AUTH_TOKEN_EXPIRED.name, ErrorCode.AUTH_TOKEN_EXPIRED.message)
        )
        response.writer.write(mapper.writeValueAsString(errorResponse))
    }
}
